packages/dumb/src/components/trends/common/focused-range-labels/focused-range-labels.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { type FunctionComponent } from 'react'
+
+import { type BgUnit } from 'medical-domain'
+
+import styles from './focused-range-labels.css'
+import { formatClocktimeFromMsPer24 } from '../../../../utils/datetime/datetime.util'
+import { type RangeSegmentSlice } from '../../../../models/enums/range-segment.enum'
+import { formatBgValue } from '../../../../utils/format/format.util'
+import Tooltip from '../../../tooltips/common/tooltip/tooltip'
+
+interface FocusedRangeLabelsProps {
+  bgUnit: BgUnit
+  focusedRangeSegments: RangeSegmentSlice
+  data: {
+    [key in RangeSegmentSlice]: number
+  } & {
+    median: number
+    msFrom: number
+    msTo: number
+    outOfRangeThresholds: {
+      low: number
+      high: number
+    }
+  }
+  position: {
+    left: number
+    tooltipLeft: boolean
+    yPositions: {
+      [key in RangeSegmentSlice]: number
+    } & {
+      topMargin: number
+    }
+  }
+}
+
+const BOTTOM_OFFSET = -5
+const TOP_OFFSET = 5
+const TOP_SIDE = 'top'
+const BOTTOM_SIDE = 'bottom'
+const TRANSPARENT_COLOR = 'transparent'
+
+export const FocusedRangeLabels: FunctionComponent<FocusedRangeLabelsProps> = (props) => {
+  const { focusedRangeSegments, data, position, bgUnit } = props
+
+  if (focusedRangeSegments.length <= 1) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Is it a good pattern to have a component return `null`? 

I feel like it would be best to check on the parent level whether to display the component or not. It's already done with the conditions `if (_.isEmpty(userTrendsState) || !userTrendsState.focusedCbgSlice || !userTrendsState.focusedCbgSliceKeys)` in `trends.js`, maybe `focusedRangeSegments.length <= 1` could be added there?

Reply:
This is what I did at first but I changed my mind. I think that the parent should indeed check wether or not the component should be displayed. But this case is also really specific and is more like a "safeguard" that is not supposed to happen. Putting this really specific logic in the parent component would be less understandable than putting it in the children (IMO).
I also wanted to directly pass `userTrendsState.focusedCbgSliceKeys[0]]` and | `userTrendsState.focusedCbgSliceKeys[1]` as props respectively named something like `bottomRangeSegment` and `topRangeSegment` and also found it harder to understand. I considered that I was simply moving logic, making either one of the component more complex, and I decided that this logic is more specific to the child component. 

This is my take on this subject! I am open to other opinions as I struggled on this, not knowing what was best.